tests/src/debug.ts

```diff
@@ -0,0 +1,693 @@
+     *
+     * @example cy.addConditionalBreakpoint(10, "NUMB = p'5'")
+     */
+    addConditionalBreakpoint(lineNumber: number, expression: string): Chainable<any>;
+  }
+}
+Cypress.Commands.add('addConditionalBreakpoint', (lineNumber: number, expression: string) => {
+  const lineNumberRegex = RegExp(`^${lineNumber}$`);
+  cy.get('.line-numbers')
+    .contains(lineNumberRegex)
+    .parent()
+    .rightclick('left', { force: true })
+    .get('.p-Menu-itemLabel')
+    .contains('Add Conditional Breakpoint...')
+    .click();
+  cy.get('.zone-widget-container.theia-debug-breakpoint-widget')
+    .find('.theia-debug-breakpoint-input')
+    .type(expression)
+    .type('{enter}');
+});
+
+declare namespace Cypress {
+  interface Chainable<Subject> {
+    /**
+     * Edit a breakpoint at specific line "Hit Count"
+     *
+     * @example cy.editBreakpointHitCount(10, "NUMB = p'5'")
+     */
+    editBreakpointHitCount(lineNumber: number, expression: string): Chainable<any>;
+  }
+}
+Cypress.Commands.add('editBreakpointHitCount', (lineNumber: number, expression: string) => {
+  const lineNumberRegex = RegExp(`^${lineNumber}$`);
+  cy.get('.line-numbers')
+    .contains(lineNumberRegex)
+    .parent()
+    .rightclick('left', { force: true })
+    .get('.p-Menu-itemLabel')
+    .contains('Edit Breakpoint...')
+    .click();
+  cy.get('.theia-debug-breakpoint-select select').select('Hit Count');
+  cy.get('.zone-widget-container.theia-debug-breakpoint-widget')
+    .find('.theia-debug-breakpoint-input')
+    .type(expression)
+    .type('{enter}');
+});
+
+declare namespace Cypress {
+  interface Chainable<Subject> {
+    /**
+     * Edit a breakpoint at specific line "Log Message"
+     *
+     * @example cy.editBreakpointLogMessage(10, "I am here with FACT: {FACT} and NUM: {NUM}")
+     */
+    editBreakpointLogMessage(lineNumber: number, expression: string): Chainable<any>;
+  }
+}
+Cypress.Commands.add('editBreakpointLogMessage', (lineNumber: number, expression: string) => {
+  const lineNumberRegex = RegExp(`^${lineNumber}$`);
+  cy.get('.line-numbers')
+    .contains(lineNumberRegex)
+    .parent()
+    .rightclick('left', { force: true })
+    .get('.p-Menu-itemLabel')
+    .contains('Edit Breakpoint...')
+    .click();
+  cy.get('.theia-debug-breakpoint-select select').select('Log Message');
+  cy.get('.zone-widget-container.theia-debug-breakpoint-widget')
+    .find('.theia-debug-breakpoint-input')
+    .type(rawTypeString(expression), { delay: 100 })
+    .type('{enter}');
+});
+
+declare namespace Cypress {
+  interface Chainable<Subject> {
+    /**
+     * Click on 'Continue'
+     *
+     * @example cy.clickOnContinue()
+     */
+    clickOnContinue(): Chainable<any>;
+  }
+}
+Cypress.Commands.add('clickOnContinue', () => {
+  cy.get(`[id^="debug:toolbar"] .debug-action.theia-debug-continue`)
+    .should('not.have.class', 'theia-mod-disabled')
+    .click()
+    .click()
+    .wait(2000);
```

Comment:
You can extract it into a separate method named `theiaClick`.

Reply:
Nice idea 👍